### ISSUE TITLE
what makes the results of code in “examples/gcn.py” random?

### ISSUE THREAD

**Issue by zhangyang1580** [rusty1s/pytorch_geometric#635]
## ❓ Questions & Help
Hi! I just takes example code ’gcn.py‘ to implement node classification. But I have **different results** every time I run the code. And in source code "gcn_conv.py", I don't find anywhere used random seed. You use  function 'glorot()' and 'zeros()' to initialize weight and bias, they are also not relate to random. What’s make the random results？
But when I add  `torch.manual_seed(5)` in the code, I get **fixed results**. The code is as follows.

````
import os.path as osp

import torch
import torch.nn.functional as F
from torch_geometric.datasets import Planetoid
import torch_geometric.transforms as T
from torch_geometric.nn import GCNConv, ChebConv  # noqa

dataset = 'Cora'
path = osp.join(osp.dirname(osp.realpath(__file__)), '..', 'data', dataset)
dataset = Planetoid(path, dataset, T.NormalizeFeatures())
data = dataset[0]


class Net(torch.nn.Module):
    def __init__(self):
        super(Net, self).__init__()
        self.conv1 = GCNConv(dataset.num_features, 16, cached=True)
        self.conv2 = GCNConv(16, dataset.num_classes, cached=True)
        # self.conv1 = ChebConv(data.num_features, 16, K=2)
        # self.conv2 = ChebConv(16, data.num_features, K=2)

    def forward(self):
        x, edge_index = data.x, data.edge_index
        x = F.relu(self.conv1(x, edge_index))
        x = F.dropout(x, training=self.training)
        x = self.conv2(x, edge_index)
        return F.log_softmax(x, dim=1)


device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')
model, data = Net().to(device), data.to(device)
optimizer = torch.optim.Adam(model.parameters(), lr=0.01, weight_decay=5e-4)


def train():
    model.train()
    optimizer.zero_grad()
    F.nll_loss(model()[data.train_mask], data.y[data.train_mask]).backward()
    optimizer.step()


def test():
    model.eval()
    logits, accs = model(), []
    for _, mask in data('train_mask', 'val_mask', 'test_mask'):
        pred = logits[mask].max(1)[1]
        acc = pred.eq(data.y[mask]).sum().item() / mask.sum().item()
        accs.append(acc)
    return accs


best_val_acc = test_acc = 0
for epoch in range(1, 201):
    train()
    train_acc, val_acc, tmp_test_acc = test()
    if val_acc > best_val_acc:
        best_val_acc = val_acc
        test_acc = tmp_test_acc
    log = 'Epoch: {:03d}, Train: {:.4f}, Val: {:.4f}, Test: {:.4f}'
    print(log.format(epoch, train_acc, best_val_acc, test_acc))
````

**Comment by WMF1997** [comment]
hello @zhangyang1580 :

it is glorot that makes it random. (if we omit dropout? i.e. dropout may be different in every train iteration, but it does not affect test part)



python code is shown below. the GCNConv use [inits.py](https://github.com/rusty1s/pytorch_geometric/blob/master/torch_geometric/nn/inits.py)'s glorot distribution. 

```python
# glorot random normal distribution 
def glorot(tensor):
    if tensor is not None:
        stdv = math.sqrt(6.0 / (tensor.size(-2) + tensor.size(-1)))
        # why it is random... and of course, no seed is set here. 
        tensor.data.uniform_(-stdv, stdv)
``` 

yours sincerely
@wmf1997

**Comment by zhangyang1580** [comment]
got it,thanks! I didn't noticed the code `tensor.data.uniform_(-stdv, stdv)` .
